### PR TITLE
AiiDA temp profile in `shell` and Jupyter-notebook

### PR DESCRIPTION
**Pull request by superstar54**
PR from AiiDA coding week. 
## Idea
One way to get people started with AiiDA is by offering a services-free option: i.e. just running codes with AiiDA:

1. Without RabbitMQ: i.e. just running in the Python session.
2. Without PostgreSQL: i.e. a temporary database (in memory or persistent sqlite)

## Implementation: AiiDA temp profile
This PR provides the services-free option for using AiiDA. This will create a temporary profile to:
- use the in-memory sqlite
- no RabbitMQ, and process backend set to `null`, a [polling-mechanism](https://github.com/aiidateam/aiida-core/blob/main/aiida/engine/runners.py#L292) is used to check the state of the process.

In the following two places:

- [x] Interactive terminal. Added an option `standalone` to `verdi shell`. 
```console
verdi shell --temp
```

- [x] Jupyter-notebook. Add a `standalone` parameter for the `aiida` command.
```python!
%load_ext aiida
aiida temp
```
- [ ] #TODO add `verdi run --temp`

### To improve
When the user runs the process use `submit`, it outputs `AssertionError: runner does not have a persister`. This may make user confuse. Better to
- disable the submit function in a `temp` profile.

### CODE REVIEW

File: aiida/cmdline/commands/cmd_shell.py [hunk 1]
```diff
@@ -14,13 +14,14 @@
...
-from aiida.cmdline.utils import decorators, echo
+from aiida.cmdline.utils import echo
 from aiida.cmdline.utils.shell import AVAILABLE_SHELLS, run_shell
 
 
 @verdi.command('shell')
-@decorators.with_dbenv()
+# @decorators.with_dbenv()
 @click.option('--plain', is_flag=True, help='Use a plain Python shell.')
+@click.option('--standalone', is_flag=True, help='Use the services-free standalone option.')
```
[the remark this answers is not in the document]
Comment (superstar54) [reply aiida/cmdline/commands/cmd_shell.py]: Thanks! I changed it to `temp` and updated the help message.

**Comment by unkcpz** [comment]
Hi @superstar54, thanks for the PR, this one uses sqlite, correct? So it goes after https://github.com/aiidateam/aiida-core/pull/6023?

**Comment by superstar54** [comment]
> Hi @superstar54, thanks for the PR, this one uses sqlite, correct? So it goes after #6023?

The PR uses the `SqliteTempBackend`, which is already in the `aiida-core`. I will list all the backends, and we can discuss which one to use in the meeting afternoon.

**Comment by sphuber** [comment]
With us moving towards supporting profiles that don't require any services, I think it might be easier and more consistent to just encourage people use that (e.g. the proposed `verdi blitz` to create an actual permanent profile that can store data). They can then use that profile as normal with any `verdi` commands and in interactive notebooks. I think that would be a better approach than implementing temporary profile support in various places. Do you agree @superstar54 ?

**Pull request closed without merging**